nixos/lib/utils.nix

```diff
@@ -54,6 +54,11 @@ rec {
     else
       shell;
 
+  # Check if the shellName appears as pname of a shell set in users.users of the given config
+  usedAsShell = shellName: config:
+    any (s: shellName == (s.pname or null))
```

Comment:
I think a better option would be to check `s.shellPath`. This is less likely to pick up something by accident, but not as strict as checking for equality on the derivation, i.e. overriding the shell package would still work.